deepmd/tf/env.py

```diff
@@ -75,17 +76,27 @@ def dlopen_library(module: str, filename: str):
     dlopen_library("nvidia.cusparse.lib", "libcusparse.so*")
     dlopen_library("nvidia.cudnn.lib", "libcudnn.so*")
 
+
+FILTER_MSGS = [
+    "is deprecated and will be removed in a future version.",
+    "disable_mixed_precision_graph_rewrite() called when mixed precision is already disabled.",
+]
+
+
+class TFWarningFilter(logging.Filter):
+    def filter(self, record):
+        return not any(msg in record.getMessage().strip() for msg in FILTER_MSGS)
+
+
 # keras 3 is incompatible with tf.compat.v1
 # https://keras.io/getting_started/#tensorflow--keras-2-backwards-compatibility
 # 2024/04/24: deepmd.tf doesn't import tf.keras any more
 
 # import tensorflow v1 compatability
-try:
-    import tensorflow.compat.v1 as tf
+import tensorflow.compat.v1 as tf
 
-    tf.disable_v2_behavior()
-except ImportError:
-    import tensorflow as tf
+tf.get_logger().addFilter(TFWarningFilter())
+tf.disable_v2_behavior()
```

Comment:
Is it necessary to remove this try&except ?

Reply:
`tf.compat.v1` was added in TF 1.13. We have dropped support for earlier versions.

Reply:
Actually, some workarounds for old TF versions can also be cleaned up

https://github.com/deepmodeling/deepmd-kit/blob/d470805170deae38ae5cccd6aae3f23db966580c/deepmd/tf/train/run_options.py#L226-L227